src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -75,6 +75,25 @@ internal Symbol GetWellKnownTypeMember(WellKnownMember member)
             return _lazyWellKnownTypeMembers[(int)member];
         }
 
+        /// <summary>
+        /// Lookup well-known member declaration in provided type.
+        ///
+        /// If a well-known member of a generic type instantiation is needed use this method to get the corresponding generic definition and
+        /// <see cref="MethodSymbol.AsMember"/> to construct an instantiation.
+        /// </summary>
+        /// <param name="type">Type that we'll try to find member in.</param>
+        /// <param name="relativeMember">A reference to a well-known member type descriptor. Note however that the type in that descriptor is ignored here.</param>
+        internal Symbol GetWellKnownTypeMember(NamedTypeSymbol type, WellKnownMember relativeMember)
```

Comment:
Using CSharpCompilation.SpecialMembersSignatureComparer.Instance should be sufficient for our purposes, We should be able to use this method during metadata import when notion of WellKnownTypes doesn't exist, but SpecialTypes are well understood. Also, we should come up with a different name for the method in order to make it clear that it is not looking for members in well known types. Something like GetRuntimeMember and call the relativeMember  parameter targetNameAndSignature or something similar.